src/core/broker_config.cpp

```diff
@@ -17,37 +17,207 @@
+// }
+bool ValidateAndStoreUid(const json& val, BrokerConfig& config)
+{
+  if (val.contains("type") && val["type"].is_string() && val.contains("manufacturer_id") &&
+      val["manufacturer_id"].is_number_integer())
   {
-    "cid": "",
-    "uid_type": "dynamic",
-    "uid": "",
+    std::string type = val["type"];
+    int64_t manufacturer_id = val["manufacturer_id"];
 
-    "dns_sd": {
-      "service_instance_name": "",
-      "manufacturer": "ETC",
-      "model": "RDMnet Broker",
-      "scope": ""
-    },
+    if (manufacturer_id > 0 && manufacturer_id < 0x8000)
+    {
+      if (type == "static" && val.contains("device_id") && val["device_id"].is_number_integer())
+      {
+        int64_t device_id = val["device_id"];
+        if (device_id >= 0 && device_id <= 0xffffffff)
+        {
+          RdmUid static_uid;
+          static_uid.manu = static_cast<uint16_t>(manufacturer_id);
+          static_uid.id = static_cast<uint32_t>(device_id);
+          config.settings.SetStaticUid(static_uid);
+          return true;
+        }
+      }
+      else if (type == "dynamic" && !val.contains("device_id"))
+      {
+        config.settings.SetDynamicUid(static_cast<uint16_t>(manufacturer_id));
+        return true;
+      }
+    }
+  }
+  return false;
+}
 
-    "max_connections": 0,
-    "max_controllers": 0,
-    "max_controller_messages": 500,
-    "max_devices": 0,
-    "max_device_messages": 500,
-    "max_reject_connections": 1000
+// Store a generic string from the "dns_sd": {} object.
+template <size_t TruncationSize>
+bool ValidateAndStoreDnsString(const json& val, std::string& string)
+{
+  std::string str_val = val;
+  if (!str_val.empty())
+  {
+    string = str_val.substr(0, TruncationSize - 1);
+    return true;
   }
-)";
+  return false;
+}
 
-BrokerConfig::BrokerConfig()
+// Validate an arithmetic type and set it in the settings struct if it is within the valid range
+// for its type.
+template <class T>
+bool SetIfInRange(const json& val, T& setting)
 {
-  defaults_ = json::parse(kDefaultSettings);
+  int64_t int_val = val;
+  if (int_val >= std::numeric_limits<T>::min() && int_val <= std::numeric_limits<T>::max())
+  {
+    setting = static_cast<T>(int_val);
+    return true;
+  }
+  return false;
 }
 
+// A typical configuration file looks something like:
+// {
+//   "cid": "4958ac8f-cd5e-42cd-ab7e-9797b0efd3ac",
+//   "uid": {
+//     "type": "dynamic",
+//     "manufacturer_id": 25972
+//   },
+//
+//   "dns_sd": {
+//     "service_instance_name": "My ETC RDMnet Broker",
+//     "manufacturer": "ETC",
+//     "model": "RDMnet Broker",
+//     "scope": "default"
+//   },
+//
+//   "max_connections": 20000,
+//   "max_controllers": 1000,
+//   "max_controller_messages": 500,
+//   "max_devices": 20000,
+//   "max_device_messages": 500,
+//   "max_reject_connections": 1000
+// }
+// clang-format off
+static const std::vector<Validator> kSettingsValidatorArray = {
+  {
+    "/cid"_json_pointer,
```

Comment:
Obviously this works, but I'm not understanding the "/cid"_json_pointer syntax.

Reply:
Here is a decent description of JSON pointers: http://rapidjson.org/md_doc_pointer.html
The `_json_pointer` suffix is a user-defined literal suffix from the library, which converts a string into its JSON pointer type.
The value of using JSON pointers is more evident if you look at the entries starting on line 152.

Reply:
I have learned new things about literal suffixes today!  Neat!  I've never used operator""